src/frontend/src/flows/addDevice/welcomeView/deviceRegistrationModeDisabled.ts

```diff
@@ -51,38 +61,28 @@ const pageContent = (userNumber: bigint) => {
     slot: pageContentSlot,
   });
 };
+
+export const deviceRegistrationDisabledInfoPage = (
+  props: Parameters<typeof deviceRegistrationDisabledInfoTemplate>[0],
+  container?: HTMLElement
+): void => {
+  const contain =
+    container ?? (document.getElementById("pageContent") as HTMLElement);
+  render(deviceRegistrationDisabledInfoTemplate(props), contain);
+};
+
 /**
  * Error page which is shown if the identy anchor does not have device registration mode enabled.
  * It shows instructions to the user on how to continue.
- * @param tentativeDeviceInfo Information about the device to be added so that the user does not have to enter everything again after enabling device registration mode.
  */
 export const deviceRegistrationDisabledInfo = (
-  connection: Connection,
-  tentativeDeviceInfo: TentativeDeviceInfo
-): void => {
-  const container = document.getElementById("pageContent") as HTMLElement;
-  render(pageContent(tentativeDeviceInfo[0]), container);
-  return init(connection, tentativeDeviceInfo);
-};
-
-const init = (
-  connection: Connection,
-  tentativeDeviceInfo: TentativeDeviceInfo
-): void => {
-  const cancelButton = document.getElementById(
-    "deviceRegModeDisabledCancel"
-  ) as HTMLButtonElement;
-
-  cancelButton.onclick = () => {
-    // TODO L2-309: do this without reload
-    window.location.reload();
-  };
-
-  const retryButton = document.getElementById(
-    "deviceRegModeDisabledRetry"
-  ) as HTMLButtonElement;
-
-  retryButton.onclick = async () => {
-    await addTentativeDevice(connection, tentativeDeviceInfo);
-  };
+  userNumber: bigint
+): Promise<"canceled" | "retry"> => {
+  return new Promise((resolve) => {
+    return deviceRegistrationDisabledInfoPage({
```

Comment:
not sure what does that particular `return`do or where it is use?

Reply:
Leftover, good catch!